### ISSUE TITLE
Path rename is overriding path filtering

### ISSUE THREAD

**Issue by jsuddsjr** [newren/git-filter-repo#110]
A check-in on May 25 added some code which changed the behavior of file filtering code. The example given on your README no longer works as expected. The subdirectory filter inadvertently causes *all* files to be included, whether they are filtered or not, and no matter if filtering is reversed with `--invert-paths`.

    git filter-repo --path src/ --to-subdirectory-filter my-module

After debugging the code, I see that path is correctly filtered and `wanted` is False. However, after path rename executes, `wanted` is set to True and the file is included anyway. I fixed the bug by adding an additional check on line 3277:

elif mod_type == 'rename' __and wanted__:

It appears that the value of `wanted` is being used inconsistently in this method.

**Comment by jsuddsjr** [comment]
PS: My hack will not work if no paths are filtered because of this code. If the `wanted` is never set, then the paths will never be renamed. Obviously, you'll need to test the expected output of `newnames()` function to ensure it behaves as expected in all cases.

````
      # If we only have renaming paths (i.e. we have no filtering paths),
      # then no path should be filtered out.  Based on how newname() works,
      # the easiest way to achieve that is setting args.inclusive to False.
      if not any(x[0] == 'filter' for x in args.path_changes):
        args.inclusive = False
````

**Comment by newren** [comment]
Eeek!!!

Well, that's very embarrassing.  I just reverted and added two more tests to make sure I don't make that same mistake again.  And I added more docs to make it clear both to myself and others the proper interaction between path filtering and path renaming.  See commits 3e1bff264c (Revert "filter-repo: fix ugly bug with mixing path filtering and renaming", 2020-06-13) and 49d6f02ff8 (filter-repo: clarify interactions between path filtering and path renaming, 2020-06-13).  Whoops.

Thanks for reporting!
